<?xml version="1.0"?>
<launch>

        <!--Load mesh model -->
        <arg name="EAI_sim_model" default="$(find dashgo_b1)/urdf/EAI_robot.urdf"/>
        <param name="robot_description" command="cat $(arg EAI_sim_model)" />

        <!--Robot state publisher -->
        <node pkg="robot_state_publisher" type="robot_state_publisher" name="b1_state_pub" />

        <!-- Uncomment the Joint state publisher if not using a node to publish joint states. If uncommented, comment any node publishing joint states-->
        <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" />


        <!--Robot Parameters-->
        <!--<arg name="wheel_radius" default="0.062"/>
        <arg name="robot_wheelbase" default="0.33"/>-->

        <!--Simulation Parameters-->
        <!--<arg name="sim_sample_time" default="0.02"/>
        <arg name="sim_node_rate" default="100"/>-->

        <!--Robot Iinitial position-->
        <!--<arg name="pos_x0" default="0.0"/>
        <arg name="pos_y0" default="0.0"/>
        <arg name="pos_th0" default="0.0"/>-->

        <!--Initialise kinematic model -->
        <!--<node pkg="dashgo_b1" type="nonholo_kinematic_sim.py" name="DashgoB1_Robot" >
            <param name="wheel_radius" value="$(arg wheel_radius)"/>
            <param name="robot_wheelbase" value="$(arg robot_wheelbase)"/>
            <param name="sim_sample_time" value="$(arg sim_sample_time)"/>
            <param name="sim_node_rate" value="$(arg sim_node_rate)"/>
            <param name="pos_x0" value="$(arg pos_x0)"/>
            <param name="pos_y0" value="$(arg pos_y0)"/>
            <param name="pos_th0" value="$(arg pos_th0)"/>
        </node> -->

        <!--Initialise joint pub node -->
        <!--<node pkg="dashgo_b1" type="b1_JointPub.py" name="Joint_Publisher" output="screen" >
            <param name="wheel_radius" value="$(arg wheel_radius)"/>
            <param name="robot_wheelbase" value="$(arg robot_wheelbase)"/>
            <param name="sim_sample_time" value="$(arg sim_sample_time)"/>
            <param name="sim_node_rate" value="$(arg sim_node_rate)"/>
        </node> -->

        <!--Initialise RVIZ -->
        <arg name="rvizconfig" default="$(find dashgo_b1)/rviz/b1_sim.rviz" />
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" /> 


        <!--Initialise teleoperation node -->
        <!--<node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="Teleop" output="screen" /> -->

</launch>